stock_intercompany/models/stock_picking.py

```diff
@@ -0,0 +1,27 @@
+from odoo import api, fields, models
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+
+    #override of method from stock module
+    def button_validate(self):
+        companies = self.env['res.company'].sudo().search([])
+        partners = {cp.partner_id: cp for cp in companies}
+
+        for picking in self:
+            if picking.partner_id in partners.keys():
+                company = partners[picking.partner_id]
+                warehouse = self.env["stock.warehouse"].sudo().search([('company_id','=',company.id)], limit=1)
+                vals = {
+                    'company_id': company.id,
+                    'picking_type_id': company.intercompany_in_type_id.id  or warehouse.in_type_id.id,
+                    'state':'draft',
+                    'location_id': self.env['stock.location'].sudo().search([('name','=','Vendors')]).id,
```

Comment:
@yvaucher  not sure about this location, there is Vendor location but no company_id on it.

Reply:
It's fine partner location are generic.
Would be better to do it as done in `stock` module. You can rely on the XML ids.

As done in:
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_picking.py#L177-L182